pkgs/applications/graphics/fiji/default.nix

```diff
@@ -0,0 +1,80 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, autoPatchelfHook
+, jdk11
+, makeDesktopItem
+, runtimeShell
+}:
+let
+  pname = "fiji";
+  version = "20201104-1356";
+
+  niceName = "Fiji Is Just ImageJ";
+
+  desktopItem = makeDesktopItem {
+    name = "fiji";
+    exec = "fiji %F";
+    icon = "fiji";
+    mimeType = "image/*;";
+    comment = "Scientific Image Analysis";
+    desktopName = niceName;
+    genericName = niceName;
+    categories = "Education;Science;ImageProcessing;";
+    terminal = false;
+    startupNotify = true;
+    extraEntries = ''
+      Version=1.0
+      TryExec=fiji
+      X-GNOME-FullName=${niceName}
+      StartupWMClass=fiji-Main
+    '';
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchurl {
+    url = "https://downloads.imagej.net/${pname}/archive/${version}/${pname}-nojre.tar.gz";
+    sha256 = "1jv4wjjkpid5spr2nk5xlvq3hg687qx1n5zh8zlw48y1y09c4q7a";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ autoPatchelfHook makeWrapper ];
+  buildInputs = [ stdenv.cc.cc.lib ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,fiji,share/pixmaps}
+
+    cp -R * $out/fiji
+    rm -f $out/fiji/jars/imagej-updater-*.jar
+
+    # Disgusting hack to stop a local desktop entry being created
+    makeWrapper ${runtimeShell} $out/bin/fiji \
+      --prefix PATH : ${lib.makeBinPath [ jdk11 ]} \
+      --set JAVA_HOME ${jdk11.home} \
+      --add-flags "-c 'exec \$($out/fiji/ImageJ-linux64 --dry-run "\$@")'"
+
+    ln $out/fiji/images/icon.png $out/share/pixmaps/fiji.png
+    ln -s "${desktopItem}/share/applications" $out/share
```

Comment:
Please use copyDesktopItems instead.

Reply:
I did not know that existed - done, and will update my other PRs.